### PR TITLE
feat: add `@scalar/api-reference-editor`

### PR DESCRIPTION
**Pull request by geoffgscott**
This adds in an editable version of the references (ESM only). 

You can import the vue component OR mountApiReferenceEditable to create an editable version of the references. For CDN delivery we may want to consider bundling Vue with the package. Otherwise import maps would have to be used (which is also well supported). 

PR also includes a very light refactor to the way Layouts are imported into APIReference. This provides a single component that can be used.

### CODE REVIEW

File: packages/api-reference-editor/src/components/ApiReferenceEditor.vue [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  </component>
+  <Layouts
+    :configuration="configuration"
+    :isDark="isDark"
+    :parsedSpec="parsedSpec"
+    :rawSpec="rawSpec"
+    @toggleDarkMode="() => toggleDarkMode()"
+    @updateContent="handleContentChange">
+    <template #editor>
+      <!--  -->
```
Comment (hwkr) [inline packages/api-reference-editor/src/components/ApiReferenceEditor.vue:152]: Intentional empty comment?

File: packages/api-reference-editor/src/components/ApiReferenceEditor.vue [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    </template>
+    <!-- Expose the content end slot as a slot for the footer -->
+    <template #content-end><slot name="footer" /></template>
+  </Layouts>
+</template>
+<style>
+body {
+  margin: 0;
+}
+</style>
```
Comment (hanspagel) [inline packages/api-reference-editor/src/components/ApiReferenceEditor.vue:166]: That’s part of the theme anyway, isn’t it? I think we can remove it.

File: packages/api-reference-editor/test/main.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  { useExternalState: true },
+  (v: string) => {
+    updateSpecValue(v)
+    console.log('input')
+  },
+)
+
+setTimeout(
+  () =>
+    updateSpecValue(`openapi: 3.0.3
```
Comment (hanspagel) [inline packages/api-reference-editor/test/main.ts:14]: Maybe we can use the new Scalar galaxy example here, usage examples here:

https://github.com/scalar/scalar/pull/1539 (was merged just now)

**Comment by hanspagel** [comment]
Did a rebase, added a demo and some minor fixes. Needs some clean up to get CI green, but it’s pretty close I’d say. ✌️

**Pull request closed without merging**